### PR TITLE
Daniel backend api

### PR DESCRIPTION
**Pull request by dpark001**
1. MoviesController seems to be working as intended. Call to API will include nextCursor to get more results on refresh with same user input.
2. DatabaseController file added. Start connecting database mapping to MoviesController.

### CODE REVIEW

**Review by emazorra — APPROVED** [review]
Looks good!

File: backend/controllers/moviesController.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+
+
+// const services = 'netflix, prime.subscription';
```
Comment (naj89) [inline backend/controllers/moviesController.js]: please remove extra comments

File: backend/controllers/moviesController.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    // const genres = '10749,35';
+    // const nextCursor = '113478:조선명탐정: 흡혈괴마의 비밀';
+    
+    const { genres, nextCursor } = req.params;
+    const services = 'netflix,prime.subscription,prime.rent,prime.buy,apple.rent,apple.buy,hbo,hulu.addon.hbo,prime.addon.hbomaxus,hulu.subscription,hulu.addon.hbo,apple.addon,peacock.free';
+    // const cursor = `&cursor=${nextCursor}`;
+    let cursor;
+    if (nextCursor) cursor = `cursor=${nextCursor}`;
+
+    const url = `https://streaming-availability.p.rapidapi.com/search/filters?services=${services}&country=us&output_language=en&order_by=original_title&genres=${genres}&genres_relation=or&desc=true&show_type=movie&${cursor}`;
```
Comment (naj89) [inline backend/controllers/moviesController.js]: optional - but for best practices I would move all params outside the url. If we decide to use any of those to filter as a feature. 
When writing code you should always think of ways to make your code easier to adjust.

File: backend/routes/movies.js [hunk 3]
```diff
@@ -1,8 +1,9 @@
 const express = require('express');
+const MoviesController = require('../controllers/moviesController');
 const router = express.Router();
 
-router.get('/', (req, res) => {
-  res.json({ message: 'Hello from movies api!' });
+router.get('/', MoviesController.getByFilters, (req, res) => {
```
Comment (naj89) [inline backend/routes/movies.js]: you need to pass a param in your get request ex. router.get('/:genreId', moviesController ....)

File: backend/routes/movies.js [hunk 4]
```diff
@@ -1,8 +1,9 @@
 const express = require('express');
+const MoviesController = require('../controllers/moviesController');
```
Comment (naj89) [inline backend/routes/movies.js:2]: controller name should be lower case

File: backend/controllers/moviesController.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+const options = {
+  method: 'GET',
+  headers: {
+    'X-RapidAPI-Key': 'f267b612b6msh504ab02ca095331p142270jsn257fad83de1e',
+    'X-RapidAPI-Host': 'streaming-availability.p.rapidapi.com'
+  }
+};
+
+const MoviesController = {
```
Comment (naj89) [inline backend/controllers/moviesController.js]: i believe controller name should be lower case

**Review by naj89 — CHANGES_REQUESTED** [review]

File: backend/controllers/moviesController.js (hunk 1 above)
Comment (dpark001) [reply backend/controllers/moviesController.js]: removed extra comments

File: backend/controllers/moviesController.js (hunk 2 above)
Comment (dpark001) [reply backend/controllers/moviesController.js]: i moved all params into their own constants, but only the filters we're actively setting are destructured/grabbed from req.params

File: backend/routes/movies.js (hunk 3 above)
Comment (dpark001) [reply backend/routes/movies.js]: rewrote route to /:genreId, i can add more params if needed

File: backend/routes/movies.js (hunk 4 above)
Comment (dpark001) [reply backend/routes/movies.js:2]: changed

File: backend/controllers/moviesController.js (hunk 5 above)
Comment (dpark001) [reply backend/controllers/moviesController.js]: changed

**Review by naj89 — APPROVED** [review]

**Pull request merged**